File: solutions/python/sieve/1/sieve.py
def primes(limit):
    is_prime = [True] * (limit - 1)

    for i in range(2, limit + 1):
        if not is_prime[i - 2]:
            continue
        
        for n in range(i, limit + 1, i):
            if n == i:
                continue
            is_prime[n - 2] = False

    return list(p for p in range(2, limit + 1) if is_prime[p - 2])

File: solutions/go/sieve/1/sieve.go
package sieve

// Sieve returns the all primes up to the given limit.
func Sieve(limit int) []int {
	primes := make([]int, 0)
	isComposite := make([]bool, limit-1)

	for i := 2; i*i <= limit; i++ {
		if !isComposite[i-2] {
			for j := i * i; j <= limit; j += i {
				isComposite[j-2] = true
			}
		}
	}

	for idx, b := range isComposite {
		if !b {
			primes = append(primes, idx+2)
		}
	}

	return primes
}
